vm/systemSmartContracts/staking.go

```diff
@@ -659,12 +675,14 @@ func (s *stakingSC) unStake(args *vmcommon.ContractCallInput) vmcommon.ReturnCod
 		return vmcommon.Ok
 	}
 
-	addOneFromQueue := !s.flagCorrectLastUnjailed.IsSet() || s.canStakeIfOneRemoved()
-	if addOneFromQueue {
-		_, err = s.moveFirstFromWaitingToStaked()
-		if err != nil {
-			s.eei.AddReturnMessage(err.Error())
-			return vmcommon.UserError
+	if !s.flagStakingV4.IsSet() {
+		addOneFromQueue := !s.flagCorrectLastUnjailed.IsSet() || s.canStakeIfOneRemoved()
```

Comment:
this file will need one big cleanup - like stakingV2 separate file and stakingV4 separate. Add a task in JIRA and we will see later.